host/matching_simulation_test.go

```diff
@@ -267,21 +270,24 @@ func (s *MatchingSimulationSuite) generate(
 	matchingClient MatchingClient,
 	domainID, tasklist string,
 	maxTasksToGenerate int,
-	taskGenerateInterval time.Duration,
+	rateLimiter *rate.Limiter,
 	generatedTasksCounter *int32,
 	lastTaskScheduleID *int32,
 	wg *sync.WaitGroup) {
 	defer wg.Done()
 
-	t := time.NewTicker(taskGenerateInterval)
-	defer t.Stop()
-
 	for {
 		select {
 		case <-ctx.Done():
```

Comment:
Nit: I think we can remove this since we handle checking if the context is done via the rateLimiter.Wait call.